tests/NoSinglePatternCaseTest.elm

```diff
@@ -11,61 +11,598 @@ all =
         [ allowed, disallowed ]
 
 
+allowed : Test
+allowed =
+    describe "allows"
+        [ test "multi-pattern cases" <|
+            \() ->
+                """module A exposing (..)
+
+type AOrB = A | B
+
+isA : AOrB -> Bool
+isA aOrB =
+    case aOrB of
+        A -> True
+        B -> False
+"""
+                    |> Review.Test.run (rule alwaysFixInArgument)
+                    |> Review.Test.expectNoErrors
+        ]
+
+
 disallowed : Test
 disallowed =
     describe "does not allow"
-        [ test "using a case to destructure" <|
+        [ withUselessPattern
+        , destructureTheArgumentTest
+        , destructureInExistingLetsTest
+        ]
+
+
+withUselessPattern : Test
+withUselessPattern =
+    describe "useless pattern (_ or ())"
+        [ describe "replace the argument"
+            [ describe "possible"
+                [ test "_" <|
+                    \() ->
+                        """module A exposing (..)
+
+always2 : a -> Int
+always2 a =
+    case a of
+        _ -> 2
+"""
+                            |> Review.Test.run (rule alwaysFixInArgument)
+                            |> Review.Test.expectErrors
+                                [ error """case a of
+        _ -> 2""" |> Review.Test.whenFixed """module A exposing (..)
+
+always2 : a -> Int
+always2 _ =
+    2
```

Comment:
Just want to point out that [`Simplify`](https://package.elm-lang.org/packages/jfmengels/elm-review-simplify/latest/Simplify) already makes this change if the pattern does not introduce any variable.